test/PR_test/integration_test/dataset/test_interleave_dataset.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2023 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import unittest
+
+import fastestimator as fe
+from fastestimator.dataset.interleave_dataset import InterleaveDataset
+from fastestimator.dataset.numpy_dataset import NumpyDataset
+from fastestimator.op.numpyop import Batch, NumpyOp
+
+
+class Plus(NumpyOp):
+    def forward(self, data, state):
+        return data + 0.5
+
+
+class TestInterleaveDataset(unittest.TestCase):
```

Comment:
should probably have a test to cover restore wizard, like in TestBatchDataset

Reply:
added 👍